### PR TITLE
Fix(eos_designs): Do not filter AVT on HA device if one path-group is present on peer

### PR DESCRIPTION
**Pull request by gmuloc**
## Change Summary

Until now, AVD is filtering completely a VT if none of the path-group listed is present locally,
This works as expected but there is one use case where this breaks potential connectivity:

Consider an HA site with two routers, `router1` connected to INET only and `router2` connected to MPLS only.
If some matched traffic is configured to be sent only on MPLS (let's say voice), then AVD would not generate the profile on `router1` for this MPLS only traffic.
When traffic for voice reaches `router1`, it will not be matched correctly and may be matched in one of the next application profile, default, or if no default is present, could be dropped.
The correct behavior is to configure the profile targeting a load-balance policy containing only the LAN_HA path-group so that the traffic can be sent to the peer and forwarded out of MPLS.

## Related Issue(s)

Reported during testing by AVD maintainers

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes

This PR adds a check where if for a given VT, no path-group is present locally but any path-group is present on the HA peer, then AVD will now generate a load-balancing policy with only the `LAN_HA` path group and will generate the profile.

## How to test

Molecule - some molecule tests were already demonstrating this.

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: python-avd/pyavd/_eos_designs/structured_config/network_services/utils.py [hunk 1]
```diff
@@ -199,6 +199,10 @@ def _configure_bgp_mlag_peer_group(self: AvdStructuredConfigNetworkServices) ->
 
         return False
 
+    ################
```
Comment (ClausHolbechArista) [inline python-avd/pyavd/_eos_designs/structured_config/network_services/utils.py]: Maybe a separate mixin?

**Review by ClausHolbechArista — APPROVED** [review]
Nice!

File: python-avd/pyavd/_eos_designs/structured_config/network_services/utils.py (hunk 1 above)
Comment (gmuloc) [reply python-avd/pyavd/_eos_designs/structured_config/network_services/utils.py]: yeah probably a btter plan it starts to grow a bit big

**Review by carlbuchmann — APPROVED** [review]
LGTM

**Pull request merged**